elytron/src/main/java/org/wildfly/extension/elytron/RoleMapperDefinitions.java

```diff
@@ -94,6 +100,24 @@
         .setRestartAllServices()
         .build();
 
+    static final SimpleMapAttributeDefinition ROLE_MAPPING_MAP = new SimpleMapAttributeDefinition.Builder(ElytronDescriptionConstants.ROLE_MAP, ModelType.LIST, false)
+            .setMinSize(1)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition KEEP_MAPPED = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.KEEP_MAPPED, ModelType.BOOLEAN, true)
```

Comment:
Are both booleans needed?  By having two boolean values we end up with four permutations, do they all definitely make sense?

Reply:
KEEP_MAPPED can be omitted, as we can define the map as:
`originalRole -> newRole,originalRole`
Both options can be replaced by using `logical-role-mapper` - I would suggest removing both of them.

Reply:
I think however keeping one does make sense, relying on a logical role mapper is actually quite a bit more configuration compared to a single boolean. - Just want to be sure four permutations make sense but as you say we can actually eliminate one of these quite easy by mapping back to the same role.

Reply:
I personally like idea of full control of logic in mapper itself. I think all of this four permutation can be useful in some situations.

At least legacy solution provide 2 permutation of them https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html-single/login_module_reference/#rolemapping_login_module

In situation when we face complaints about Elytron beeing difficult to configure I wouldn't complicate things with another resource (logical-role-mapper) if it is not really necessary.